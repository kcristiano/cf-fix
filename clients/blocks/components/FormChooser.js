const SelectControl = wp.components.SelectControl;
import {CALDERA_FORMS_STORE_NAME} from "../store";
const { __ } = wp.i18n;

/**
 * Basic component to choose forms with
 *
 * @param props
 * @return {XML}
 * @constructor
 */
export const FormChooser = (props) => {
    return (
        <SelectControl
            className={'caldera-forms-form-chooser'}
            label={ __( 'Form' ) }
            value={ props.formId }
            options={ props.forms.map( (form) => ( {
                value: form.formId,
                label: form.name,
            } ) ) }
            onChange={ (newValue) => {props.onChange(newValue)} }
        />
    )
};

//Import wp.data's HOC
const { withSelect } = wp.data;

/**
 * Form chooser wrapped in form selector
 */
export const FormChooserWithSelect = withSelect( (select, ownProps ) => {
    const { getForms } = select( CALDERA_FORMS_STORE_NAME);
    return {
        forms: getForms()
    };
} )( FormChooser );







