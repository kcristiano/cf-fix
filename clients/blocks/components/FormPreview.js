import {CALDERA_FORMS_STORE_NAME} from "../store";
import {appendAssets} from "../../functions/appendAssets";
const { withSelect } = wp.data;
const Spinner = wp.components.Spinner;

function createMarkup(preview) {
    return {__html: preview};
}

/**
 * Render a  form preview
 *
 * @since 1.6.2
 *
 * @param props
 * @return {XML}
 * @constructor
 */
export const FormPreview = (props) => {
    const id = 'caldera-forms-form-preview';
    if (undefined !== props.preview && props.preview.html ) {
        appendAssets(props.preview.css, props.preview.js);
        return <div id={id} dangerouslySetInnerHTML={createMarkup(props.preview.html)}/>;
    } else {
        return <div id={id}><Spinner /></div>;
    }
};


/**
 * Wrap with data selectors
 */
export const FormPreviewWithSelect = withSelect( (select, ownProps ) => {
    const { getFormPreview } = select( CALDERA_FORMS_STORE_NAME);
    return {
        preview: getFormPreview(ownProps.formId),
    };
} )( FormPreview );

