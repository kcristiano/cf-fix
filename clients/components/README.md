# Shared Components
This is a library of components that can safely be shared by any React UI, Gutenberg or not and are not specific to a client.


## Components
* `FormSelectorNoGutenberg` A react-bootstrap dependent selector for all saved forms.
