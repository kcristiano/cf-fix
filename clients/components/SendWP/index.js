import {GetSendWP} from "./GetSendWP/GetSendWP";

export {
    GetSendWP
}