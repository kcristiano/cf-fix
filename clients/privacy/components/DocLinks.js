import React from 'react';
import { Twemoji } from 'react-emoji-render';
import {ListGroup, ListGroupItem } from 'react-bootstrap'
export const DocLinks = () => {
    return (
        <React.Fragment>
            <ListGroup
                style={{padding:'12px'}}
            >
                <ListGroupItem>
                    <Twemoji text=":eyes:" />
                    <a href="https://calderaforms.com/doc/setting-caldera-forms-gdpr-data-requests/?utm_source=wp-admin&utm_campaign=privacy-settings">
                        Setting Up Caldera Forms For GDPR Requests
                    </a>
                </ListGroupItem>
                <ListGroupItem>
                    <Twemoji text=":eyes:" />
                    <a href="https://calderaforms.com/gdpr/?utm_source=wp-admin&utm_campaign=privacy-settings">
                       All Caldera Forms GDPR Compliance Tools
                    </a>
                </ListGroupItem>
            </ListGroup>
        </React.Fragment>
    )
};

