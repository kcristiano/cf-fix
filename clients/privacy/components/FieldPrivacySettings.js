import React from  'react';
import PropTypes from 'prop-types';
import { FormGroup,FormControl,ControlLabel,HelpBlock,Panel,PanelGroup,Checkbox } from 'react-bootstrap';

export  class FieldPrivacySettings extends React.Component {

    constructor(props){
        super(props);
        this.state = {
            isIdentifying: false
        };
        this.handleCheck = this.handleCheck.bind(this);
    }

    handleCheck(){
        this.setState( { isIdentifying: ! this.state.isIdentifying } );
    };
    render(){
        return(
            <PanelGroup>
                <Panel onClick={(e) => {}}>
                    <Panel.Body>
                        <Panel.header>
                            <p>{field.label}</p>
                        </Panel.header>
                        <FormGroup controlId={`caldera-forms-privacy-is-email-identifier-field-${this.props.field.ID}`}>
                            <ControlLabel>
                                Submitter Email
                            </ControlLabel>
                            <Checkbox
                                onChange={this.handleCheck}
                                checked={this.state.isIdentifying}
                            >
                                Checkbox
                            </Checkbox>
                            <HelpBlock>Can this field be used to identify form submitter by email?</HelpBlock>}
                        </FormGroup>
                    </Panel.Body>
                    <Panel.Footer>Panel footer</Panel.Footer>
                </Panel>
            </PanelGroup>
        )
    }


}

FieldPrivacySettings.propTypes = {
    field: PropTypes.object.isRequired,
    formId: PropTypes.string.isRequired
};