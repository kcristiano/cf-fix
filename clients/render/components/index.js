import {CalderaFormsFieldRender,CalderaFormsFieldPropType} from "./CalderaFormsFieldRender";
import {CalderaFormsRender} from "./CalderaFormsRender";
import {CalderaFormsFieldGroup} from "./CalderaFormsFieldGroup";

export{
	CalderaFormsFieldGroup,
	CalderaFormsFieldPropType,
	CalderaFormsFieldRender,
	CalderaFormsRender
}