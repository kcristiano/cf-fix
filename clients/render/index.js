import {CalderaFormsRender} from "./components/CalderaFormsRender";

import React from 'react';
import ReactDOM from "react-dom";
import domReady from '@wordpress/dom-ready';

Object.defineProperty( global.wp, 'element', {
	get: () => React
} );




domReady( function() {
	jQuery( document ).on( 'cf.form.init', (e, obj ) => {
		const {
			state,//CF state instance
			formId, //Form's ID
			idAttr, //Form element id attribute,
			//$form, //Form jQuery object
		} = obj;
		const fieldsToControl = [];

		//Build configurations
		document.querySelectorAll('.cf2-field-wrapper' ).forEach(function(fieldElement) {
			const fieldIdAttr = fieldElement.getAttribute('data-field-id');

			const formConfig = window.cf2[idAttr];

			let fieldConfig = formConfig.fields.hasOwnProperty(fieldIdAttr) ?
				formConfig.fields[fieldIdAttr]
				: null;

			if( 'string' === typeof  fieldConfig ){
				fieldConfig = JSON.parse(fieldConfig);
			}
			if( fieldConfig ){
				fieldsToControl.push(fieldConfig);
				if(fieldConfig.hasOwnProperty('fieldDefault' ) ){
					state.mutateState(fieldIdAttr,fieldConfig.fieldDefault);

				}
			}
		});


		/**
		 * Flag to indicate if validation is happening or not
		 *
		 * This is controlled outside of React so that CF1 can trigger validation
		 *
		 * @since 1.8.0
		 *
		 * @type {boolean}
		 */
		let shouldBeValidating = false;

		//When form submits, push values onto form object before it is serialized
		jQuery(document).on( 'cf.form.submit', (event,obj) => {
			const values = theComponent.getFieldValues();
			if( Object.keys(values).length){
				Object.keys(values).forEach(fieldId => {
					obj.$form.data(fieldId, values[fieldId]);
				});
			}
		});

		/**
		 * Ref for rendered app
		 *
		 * @see https://reactjs.org/docs/refs-and-the-dom.html
		 *
		 * @since 1.8.0
		 * @type {*}
		 */
		let theComponent = '';
		ReactDOM.render(
			<CalderaFormsRender
				cfState={state}
				formId={formId}
				formIdAttr={idAttr}
				fieldsToControl={fieldsToControl}
				shouldBeValidating={shouldBeValidating}
			ref={(component) => {theComponent = component}} />,
			document.getElementById(`cf2-${idAttr}`)
		);


	});
} );