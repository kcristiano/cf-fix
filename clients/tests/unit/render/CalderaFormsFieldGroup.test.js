import {CalderaFormsFieldGroup} from "../../../render/components";
import renderer from 'react-test-renderer';

describe('CalderaFormsFieldGroup component', () => {
	const fieldIdAttr = 'fld_5843941_1';
	const fieldConfig = {
		fieldCaption: "",
		fieldDefault: "new default",
		fieldId: "fld_5843941",
		fieldIdAttr,
		fieldLabel: "Text 2 Field",
		fieldPlaceHolder: "",
		fieldValue: "",
		outterIdAttr: "cf2-fld_5843941_1",
		required: false,
		type: "text",
	};
	const handler = () => {
	};
	it('Matches snapshot with basic args', () => {
		expect(
			renderer.create(
				<CalderaFormsFieldGroup
					onChange={handler}
					field={fieldConfig}
				/>
			).toJSON()
		).toMatchSnapshot();
	});


	it('Does not render if shouldShow is false', () => {
		expect(
			renderer.create(
				<CalderaFormsFieldGroup
					onChange={handler}
					field={fieldConfig}
					shouldShow={false}
				/>
			).toJSON()
		).toEqual(null)
	});

	it('Creates the label correctly', () => {
		const testRenderer = renderer.create(<CalderaFormsFieldGroup
			onChange={handler}
			field={fieldConfig}
			shouldDiable={true}
		/>);
		const testInstance = testRenderer.root;

		expect(testInstance.findByType('label').props.htmlFor).toBe(fieldIdAttr);
		expect(testInstance.findByType('label').props.className).toBe('control-label');
		expect(testInstance.findByType('label').props.children).toBe(fieldConfig.fieldLabel);


	});


	it('Creates the inner input correctly', () => {
		const testRenderer = renderer.create(<CalderaFormsFieldGroup
			onChange={handler}
			field={fieldConfig}
		/>);
		const testInstance = testRenderer.root;

		expect(testInstance.findByType('input').props.required).toBe(false);
		expect(testInstance.findByType('input').props.className).toBe('form-control');
		expect(testInstance.findByType('input').props.placeholder).toBe('');
		expect(testInstance.findByType('input').props.type).toBe('text');
		expect(testInstance.findByType('input').props.value).toBe('');
		expect(typeof testInstance.findByType('input').props.onChange).toBe('function');


	});

	it('Disables is shouldDisable is true', () => {
		const testRenderer = renderer.create(<CalderaFormsFieldGroup
			onChange={handler}
			field={fieldConfig}
			shouldDisable={true}
		/>);
		const testInstance = testRenderer.root;

		expect(testInstance.findByType('input').props.disabled).toBe(true);

	});

	it('Sets required if it should', () => {
		const testRenderer = renderer.create(<CalderaFormsFieldGroup
			onChange={handler}
			field={{
				...fieldConfig,
				required: true
			}}
		/>);
		const testInstance = testRenderer.root;

		expect(testInstance.findByType('input').props.required).toBe(true);

	});

	it('Adds caption and aria attribute', () => {
		const testRenderer = renderer.create(<CalderaFormsFieldGroup
			onChange={handler}
			field={{
				...fieldConfig,
				caption: "Click Me"
			}}
		/>);
		const testInstance = testRenderer.root;
		expect(testInstance.findByType('span').props.className).toBe( 'help-block');
		expect(testInstance.findByType('span').props.children).toBe( 'Click Me');




	});

});