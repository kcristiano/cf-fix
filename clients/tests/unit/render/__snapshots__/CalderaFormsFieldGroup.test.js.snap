// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalderaFormsFieldGroup component Matches snapshot with basic args 1`] = `
<div
  className="form-group cf2-field-group"
>
  <label
    className="control-label"
    htmlFor="fld_5843941_1"
  >
    Text 2 Field
  </label>
  <input
    className="form-control"
    disabled={false}
    id="fld_5843941_1"
    onChange={[Function]}
    placeholder=""
    required={false}
    type="text"
    value=""
  />
</div>
`;
